### ISSUE TITLE
source_id native_nominal_resolution corrections

### ISSUE THREAD

**Issue by taylor13** [WCRP-CMIP/CMIP6_CVs#829]
I have reviewed the native_nominal_resolution recorded for each model.  I could only check models with latxlon grids with approximately even-spacing (i.e., gaussian or regular cartesian latxlon grids) and models using an icosahedral grid (with cells of uniform size).  

"nominal resolution" is defined in [Appendix 2](https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk/edit?ts=5da4a3dd#bookmark=id.ibeh7ad2gpdi) of the [metadata specifications document](https://goo.gl/v1drZl).  For these grids I think the following models need correcting:

BCC_AGCM3_LR  (128x64 lonxlat) has mean resolution = 402 km. nominal_resolution should be 500 km (not 250 km).

CAS-ESM1-0 (256x128 lonxlat) has mean resolution = 201 km.  nominal_resolution should be 250 km (not 100 km).

CMCC-CM2-VHR4 (1152x768) has a mean resolution = 38 km.  nominal_resolution should be 50 km (not 25 km).

NorESM2-HH (1152x768) has a mean resolution = 38 km.  nominal_resolution should be 50 km (not 25 km).

INM-CM4-8INM-CM5-0 (180x120) has a mean resolution = 245 km.  nominal_resolution should be 250 km (not 100 km).

INM-CM5-H 50 (540x360) has a mean resolution = 81 km.  nominal_resolution should be 100 km (not 50 km).

IPSL-CM6A-ATM-HR (512x360) has a mean resolution = 84 km.  nominal_resolution should be 100 km (not 50 km).

MCM-UA-1-0 (96x80) has a mean resolution = 417 km.  nominal_resolution should be 500 km (not 250 km).

MRI-AGCM3-2-H (640x320) has a mean resolution = 80 km.  nominal_resolution should be 100 km (not 50 km).

MRI-AGCM3-2-H (640x320) has a mean resolution = 80 km.  nominal_resolution should be 100 km (not 50 km).

BCC_AGCM3_MR (320x180) has mean-resolution of 160.8, which just missed the upper bound for a 100 km nominal_resolution, so they fall in the 250 km category (not 100 km)  This is not seriously misleading, but I would like it corrected.

CAMS-CSM1-0 (320x180) has mean-resolution of 160.8, which just missed the upper bound for a 100 km nominal_resolution, so they fall in the 250 km category (not 100 km)  This is not seriously misleading, but I would like it corrected.

MRI-ESM2.0 (320x180) has mean-resolution of 160.8, which just missed the upper bound for a 100 km nominal_resolution, so they fall in the 250 km category (not 100 km)  This is not seriously misleading, but I would like it corrected.

NICAM16-7S icosahedral should probably be 50 km (not 100 km)
NICAM15-8S icosahedral should probably be 25 km (not 50 km)
NICAM16-9D-L78 and NICAM16-9S icosahedral should probably be 10 km (not 25 km)

FIO-ESM-2-0  report a "0.9x1.25 finite volume grid; 192 x 288 longitude/latitude ...", but I suspect the order of lon and lat have been reversed.  Check this and make consistent with ordering in other models: probably should read "1.25x0.9 finite volume grid; 288 x 192 longitude/latitude ..."

NorCPM1, NorESM1-F, NorESM2-LM, NorESM2-LME, NorESM2-LMEC, NorESM2-MM model components and check that the grid dimensions (360 x 384) are in longitude x latitude order (and not reversed).  

GISS-E2-1-G(atmos and ocean) and the ocean component of GISS-E2-1-G-CC GISS-E2-1-H GISS-E2-2-G GISS-E3-G are labeled 100 km, which is not misleading, but they can be more precisely labeled "1x1 degree" (which is a special designation provided for the "standard grid").

**Comment by durack1** [comment]
@taylor13 chat about this next week, not 100% sure how to proceed as many groups will already have written this information into their files using the pre-configured [cmip6-cmor-table](https://github.com/PCMDI/cmip6-cmor-tables/) inputs

**Comment by taylor13** [comment]
The cmip6_CV.json file only contains a list of the allowed "nominal_resolutions", which we are not changing.  We don't transfer for each model the native_nominal_resolution to cmip6-CV.json.

I'm not suggesting that we correct the already written files (or ask groups to change the resolution recorded for a run that is in the process of being written), but I don't think there is any harm in correcting the source_id.json file.  We might as well enable users to retrieve the native_nominal_resolutions that are *correct*. 

 Not sure whether ES-DOC records native_nominal_resolution.  We should check on that.

**Comment by durack1** [comment]
@taylor13 you're right, we no longer include resolution info, see [cmip6-cmor-tables/Tables/CMIP6_CV.json](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_CV.json#L137)

**Comment by tinoue70** [comment]
@taylor13 for NICAM-* models, we calculated grid resolution from using grid point data according to the "Algorithm for defining the nominal_resolution". 

NICAM16-7S icosahedral: avg(d_max) = 72.8km -> 100 km
NICAM15-8S icosahedral: avg(d_max) = 36.4km -> 50 km
NICAM16-9D-L78 and NICAM16-9S: avg(d_max) = 18.2km -> 25 km

The distribution of the grid point is **not uniform**. A standard icosahedral grid system has two polygons: hexagon and pentagon. The size difference between these two polygons causes crucial errors in atmospheric simulations (Tomita et al., 2002). To reduce the noise, we apply the grid point modification with a spring-dynamics approach (Tomita et al. 2001, 2002). As a result, distances between each grid and the neighboring grid is not uniform and d_max is larger than we expected.

**Comment by taylor13** [comment]
thanks, @tinoue70, for checking the resolutions.  I'm confident your numbers are correct, so I went back to see why mine were different.  Mine were wrong for at least 2 reasons:
1) I assumed grid cells were circles and divided the global surface area by the number of cells to get the area of individual cells.  I then calculated grid cell diameters consistent with the area (yielding an estimate of d_max).

2)  Even if I had used the area of regular hexagons to compute d_max, I would have found the following:
NICAM16-7S icosahedral: avg(d_max) = 69.2 km
NICAM15-8S icosahedral: avg(d_max) = 34.6 km
NICAM16-9D-L78 and NICAM16-9S: avg(d_max) = 17.3 km

In the first two cases, these put you in different nominal resolution categories from the correct ones you calculated (because your resolutions are near the cutoff between two categories).  It seems that the spring-dynamics distortion of the grid cells from regular hexagons slightly increases d_max (but only by about 5%).

Thanks again for checking the nominal resolution.  We'll retain the values you originally registered with your models.

**Comment by taylor13** [comment]
to be sure, @tinoue70  .... in calculating d_max, did you measure the distances between the triangle vertices (the "P" points in fig. 3 of Tomita et al. (2001)) or the hexagon vertices (the "G" points in that same figure)?  I think it's the max distance between "G" points that we want (which I think is somewhat shorter than the distance between "P" points.

**Comment by hisashiyashiro** [comment]
I will answer on behalf of @tinoue70. We used "G" points. Specifically, we calculated the great circle distance between neighboring vertices of each hexagon or pentagon.

**Comment by taylor13** [comment]
Excellent! Thanks again @tinoue70 and @hisashiyashiro for clarifying and for your participation in CMIP6.  I'll edit the first comment at the top of this thread to summarize.

**Comment by durack1** [comment]
@taylor13 let me know when you want to proceed with these changes. We'll need to clarifying with each of the modeling group representatives that all the recommended changes above https://github.com/WCRP-CMIP/CMIP6_CVs/issues/829#issue-523072132 are valid
